### PR TITLE
Rework breaking changes section

### PR DESCRIPTION
**Pull request by gewarren**
[Preview link](https://review.docs.microsoft.com/en-us/dotnet/core/compatibility/net5-breaking-changes-overview?branch=pr-en-us-21559).

Note: This PR's base branch is a "release" branch for the breaking changes rework, not master. It would be too much to review if I converted all of the .NET 5.0 breaking changes in a single PR. Once this PR is merged, I'll work on the moving over the next chunk of breaking changes.

### CODE REVIEW

File: docs/core/compatibility/code-analysis/5.0/ca2014-stackalloc-in-loops.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
```
Comment (gewarren) [inline docs/core/compatibility/code-analysis/5.0/ca2014-stackalloc-in-loops.md:1]: This is not actually a new file, just a rename of an include and adjustment of the H1-H4 levels and relative links.

File: docs/core/compatibility/code-analysis/5.0/ca2247-ctor-arg-should-be-taskcreationoptions.md [hunk 2]
```diff
@@ -1,34 +1,39 @@
...
-- To disable code analysis completely, set `EnableNETAnalyzers` to `false` in your project file. For more information, see [EnableNETAnalyzers](../../../../docs/core/project-sdk/msbuild-props.md#enablenetanalyzers).
+- To disable code analysis completely, set `EnableNETAnalyzers` to `false` in your project file. For more information, see [EnableNETAnalyzers](../../../project-sdk/msbuild-props.md#enablenetanalyzers).
 
-#### Category
+## Category
 
 Code analysis
 
-#### Affected APIs
+## Affected APIs
```
Comment (Youssef1313) [inline docs/core/compatibility/code-analysis/5.0/ca2247-ctor-arg-should-be-taskcreationoptions.md:30]: :warning: This breaks the API port tool:

https://github.com/microsoft/dotnet-apiport/blob/19e0d616251a7416116449881dadf24b18180b89/src/lib/Microsoft.Fx.Portability/BreakingChangeParser.cs#L115

The current code searches for `###`, then looks for specific headings in the switch statement, including "category" and "affected apis".

**Comment by BillWagner** [comment]
I really like this presentation @gewarren  The way you've used an index page for each release and the TOC for area-by-area is a great innovation to make sure both use cases are handled well.

**Review by tdykstra — COMMENTED** [review]
Agree with @BillWagner , this looks good.

Should the doc ".NET Core breaking change selectors" be moved to under the ".NET Core versions" TOC node?

The "Breaking changes in .NET 5.0" overview doc says "Breaking changes are organized by category in the table of contents." -- it looks like the categories are in the 3.1-5.0 doc rather than in the toc. Actually I wonder if this Overview toc node could be removed, it doesn't seem to be essential.

**Comment by gewarren** [comment]
> Should the doc ".NET Core breaking change selectors" be moved to under the ".NET Core versions" TOC node?

Yes, thanks.

> The "Breaking changes in .NET 5.0" overview doc says "Breaking changes are organized by category in the table of contents."

They are organized primarily by .NET 5+ version and secondarily by category.

> Actually I wonder if this Overview toc node could be removed

It is pretty fluffy, but I need somewhere concrete to link to from the .NET Core versions "selectors" page for.

**Review by tdykstra — APPROVED** [review]

**Pull request merged**